scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -143,10 +143,15 @@ def run_robot_test(self, command, handler):
 
             if cmake_proc.returncode == 0:
                 self.instance.status = "passed"
+                # all tests in one Robot file are treated as a single test case,
```

Comment:
I do not know to much about robot tests, but if there is generated any report file after robot tests, it would be useful to parse that file and add testcases with statuses. Such approach is used in `Pytest._parse_report_file`. Then you can find in twister.json (or other report files) all subtests.